src/toil/batchSystems/abstractBatchSystem.py

```diff
@@ -66,11 +69,12 @@ class BatchJobExitReason(enum.IntEnum):
     """Job was not able to talk to the leader via the job store, so Toil declared it failed."""
     BADCONSTRAINTS: int = 257
     """Job landed on a node with incompatible resources"""
-    OVERUSE = 253
+    PKILL: int = 9
+    """Process was killed by user/kernel"""
```

Comment:
This conflicts `PARTITION: int = 9` defined by toil devs.
We can avoid this line. 
```suggestion
    #PKILL: int = 9
    #"""Process was killed by user/kernel"""
```

Reply:
We will change the exit code of Partition to 10, since it holds different meaning than sigkill. We will also keep Pkill as 9. We would not double the memory on Partition since it is used in the code to notify that job is not able to communicate its status to the leader. Please let me know your thoughts on this @rohith-bs

Reply:
added a new commit to treat partition as sigkill and enabled double memory for this exit_reason